ballerina/http_request.bal

```diff
@@ -414,17 +415,33 @@ public class Request {
 
     # Sets a `json` as the payload. If the content-type header is not set then this method set content-type
     # headers with the default content-type, which is `application/json`. Any existing content-type can be
-    # overridden by passing the content-type as an optional parameter.
+    # overridden by passing the content-type as an optional parameter. If the given payload is a record type with 
+    # the `@jsondata:Name` annotation, the `jsondata:toJson` function internally converts the record to JSON
     #
     # + payload - The `json` payload
     # + contentType - The content type of the payload. This is an optional parameter.
     #                 The `application/json` is the default value
     public isolated function setJsonPayload(json payload, string? contentType = ()) {
         mime:Entity entity = self.getEntityWithoutBodyAndHeaders();
-        setJson(entity, payload, self.getContentType(), contentType);
+        setJson(entity, jsondata:toJson(payload), self.getContentType(), contentType);
```

Comment:
This `jsondata:toJson` can panic right?

Reply:
According to the `toJson` method signature. it only returns `json` type there is no error return. Yeah, there can be panic happened, if the user has provided the value which will lead to some runtime error. Are we going to handle it here or keep this as it is based on the `toJson` function?

Reply:
In this specific scenario, I don't think it will panic since the input is also a json type. So it is not required to handle that panics